[18.0][MIG] product_harmonized_system_delivery: Migration to 18.0

LucasTran380381 opened the pull request:
depends: `https://github.com/LucasTran380381/intrastat-extrastat/pull/1`

alexis-via committed:
Add new module product_harmonized_system_delivery (hide native hs_code field)

Finalize port to v11

alexis-via committed:
Prepare v12 branch

alexis-via committed:
[MIG] Migrate all modules from v11 to v12

BenjaHe committed:
[13.0] [MIG] product harmonized system (#80)

* Large code re-engineering following the Akretion-Noviat code sprint on intrastat

This code has been written both by Luc de Meyer and myself.

* hs_code display_name fix

* major update intrastat V3 modules

* Add product_origin_country_id on declaration/computation lines

Copy incoterms and destination country from SO to invoice when invoicing from SO
We need weight even when supplementary units is used
Small cleanups and enhancements

* Add support for accessory costs

Add default values for intrastat transaction on company
Code cleanup

* this week's ci errors

* Update README.rst

Rename variables

* Port intrastat_base to v10

Set 2 other modules to uninstallable
Update README.rst: switch to new intrastat project

* Port product_harmonized_system to v10

* Fix warning on display_name

Special thanks to Pedro and Holger for finding the solution for display_name
Fix demo data

* [MIG] product_harmonized_system: Migration to 11.0

* [FIX] product_harmonized_system: fix get_hs_code_recurively

* [IMP] *: removed utf-8 headers, changed icon and site

* company model test, country model test as in v8
* png logo instead of svg, removed Python 2 utf-8 headers

* [UPD] Update product_harmonized_system.pot

* Add new module product_harmonized_system_delivery (hide native hs_code field)

Finalize port to v11

* [FIX+IMP] intrastat_*: Several things:

* Add readonly on some fields when state of declaration is done
* Division by zero in computation of accessory costs
* Add FR translation
* Fix strings
* Minor code updates

* [MIG] intrastat_product: Migration to 11.0

* Small fixes in intrastat_base
* account_tax_template

* [ADD] icon.png

* [UPD] Update product_harmonized_system.pot

* Prepare v12 branch

* [MIG] Migrate all modules from v11 to v12

* flake8

* [UPD] Update product_harmonized_system.pot

* Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-12.0/intrastat-extrastat-12.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-12-0/intrastat-extrastat-12-0-product_harmonized_system/

* [FIX] product_harmized_system DESCRIPTION.rst

* [UPD] README.rst

* [UPD] README.rst

* Added translation using Weblate (Spanish)

* Translated using Weblate (Spanish)

Currently translated at 50.0% (18 of 36 strings)

Translation: intrastat-extrastat-12.0/intrastat-extrastat-12.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-12-0/intrastat-extrastat-12-0-product_harmonized_system/es/

* [MIG] product_harmonized_system : Migration to 13.0

* [IMP] product_harmonized_system: black, isort

[UPD] README.rst

[UPD] Update product_harmonized_system.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-13.0/intrastat-extrastat-13.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-13-0/intrastat-extrastat-13-0-product_harmonized_system/

Commit:
[REF] product_harmonized_system_delivery: Black python code

Commit:
[MIG] product_harmonized_system_delivery: Migration to 13.0

luc-demeyer committed:
[13.0][FIX]replace deprecated _company_default_get

product_harmonized_system 13.0.1.0.1

pre-commit update

alexis-via committed:
intrastat_product: Add xml_attachment_id field

intrastat_product: add mail.activity.mixin and widget in form view
product_harmonized_system: remove archive button (use Action menu)
All modules:
- use super() instead of super(class, self)
- Update domain_force in multi-company ir.rule to v13 standard

alexis-via committed:
[MIG] product_harmonized_system to v14

[UPD] README.rst

[UPD] Update product_harmonized_system.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-product_harmonized_system/

alexis-via committed:
[MIG] product_harmonized_system_delivery to v14

Mat-moran committed:
Translated using Weblate (Spanish)

Currently translated at 80.5% (29 of 36 strings)

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-product_harmonized_system/es/

AdriaGForgeFlow committed:
[13.0] product_harmonized_system: do not require company on hs_code model

product_harmonized_system 14.0.1.0.1

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (36 of 36 strings)

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-product_harmonized_system/fr/

hbrunn committed:
[IMP] product_harmonized_system_delivery: Make product.template#hs_code related

alexis-via committed:
[MIG] hs_code_id company_dependent=True to False

Migration script provided for product.template and product.category
Fixes bug #126

alexis-via committed:
[MIG] product_harmonized_system to v15
TT36971

alexis-via committed:
product_harmonized_system_delivery: hs_code is now store=True

This commit is the "followup" of the change to company_dependent=False
on the fields hs_code_id on product.template and product.category.
Add groupby on hs_code_id on product.template search view
hs.code is now company_id=False by default

alexis-via committed:
product_harmonized_system_delivery: hs_code is now store=True

This commit is the "followup" of the change to company_dependent=False
on the fields hs_code_id on product.template and product.category.
Add groupby on hs_code_id on product.template search view
hs.code is now company_id=False by default

[UPD] Update product_harmonized_system.pot

product_harmonized_system 14.0.2.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-product_harmonized_system/

luc-demeyer committed:
[14.0]Intrastat Brexit support

luc-demeyer committed:
[14.0]intrastat - improved brexit support

[UPD] Update product_harmonized_system.pot

product_harmonized_system 14.0.2.2.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-product_harmonized_system/

yweng8111 committed:
[IMP] add tag 'sheet' into form view of hs.code

product_harmonized_system 14.0.2.3.0

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 10.5% (4 of 38 strings)

Translation: intrastat-extrastat-14.0/intrastat-extrastat-14.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-14-0/intrastat-extrastat-14-0-product_harmonized_system/hr/

alexis-via committed:
[MIG] product_harmonized_system_delivery to v15

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-15.0/intrastat-extrastat-15.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-15-0/intrastat-extrastat-15-0-product_harmonized_system_delivery/

alexis-via committed:
[MIG] product_harmonized_system to v16

alexis-via committed:
[MIG] product_harmonized_system_delivery to v16

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (35 of 35 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system/hr/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/hr/

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (35 of 35 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system/es/

Gelo-fl committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (3 of 3 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/es/

sanrav committed:
Translated using Weblate (French)

Currently translated at 100.0% (35 of 35 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system/fr/

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (35 of 35 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system/fr/

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/fr/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/es/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system_delivery
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system_delivery/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (35 of 35 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system/it/

victoralmau committed:
[FIX] product_harmonized_system: Change web_ribbon text to title to make it translatable

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (35 of 35 strings)

Translation: intrastat-extrastat-16.0/intrastat-extrastat-16.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-16-0/intrastat-extrastat-16-0-product_harmonized_system/it/

AndreuOForgeFlow committed:
[IMP] product_harmonized_system: pre-commit auto fixes

AndreuOForgeFlow committed:
[MIG] product_harmonized_system: Migration to 17.0

AndreuOForgeFlow committed:
[IMP] product_harmonized_system: solved ACL with an empty group

AndreuOForgeFlow committed:
[FIX] product_harmonized_system: Resolve AttributeError in hs_code _compute_display_name

AndreuOForgeFlow committed:
[IMP] product_harmonized_system_delivery: pre-commit auto fixes

AndreuOForgeFlow committed:
[MIG] product_harmonized_system_delivery: Migration to 17.0

Commit:
[IMP] product_harmonized_system: pre-commit execution

Commit:
[MIG] product_harmonized_system: Migration to 18.0

Commit:
[ADD] add checklog-odoo.cfg

[Fix] fix review point

Commit:
[MIG] product_harmonized_system_delivery: Migration to 18.0

[FIX] change no-stored field

[ADD] add contributor

nguyenminhchien commented on product_harmonized_system_delivery/models/product_template.py:
```diff
@@ -8,5 +8,5 @@ class ProductTemplate(models.Model):
-    hs_code = fields.Char(related="hs_code_id.hs_code", store=True)
-    country_of_origin = fields.Many2one(related="origin_country_id", store=True)
+    hs_code = fields.Char(related="hs_code_id.hs_code")
+    country_of_origin = fields.Many2one(related="origin_country_id")
```
Do you have a good reason for removing store=True from these 2 fields?

LucasTran380381 replied:
In Odoo 18, it's true that you don't necessarily need to store a field to search for it. Odoo's ORM (Object-Relational Mapping) allows you to search on computed fields without storing them.

nguyenminhchien replied:
yes, but it's bad for performance compared to store=True. Please keep as it is

nguyenminhchien reviewed (changes requested):
Please keep the commit title following the naming convention: [MIG] $module-name: Migration to 18.0.
You can add these info in your PR's description
```
[FIX] change no-stored field
[ADD] add contributor
```
![image](https://github.com/user-attachments/assets/97742df2-f8b3-432a-a00e-dd97525affb8)

nguyenminhchien reviewed (changes requested):
check my comment

OCA-git-bot committed:
Merge PR #288 into 18.0

Signed-off-by StefanRijnhart

Commit:
[UPD] Update product_harmonized_system.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: intrastat-extrastat-18.0/intrastat-extrastat-18.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-18-0/intrastat-extrastat-18-0-product_harmonized_system/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (34 of 34 strings)

Translation: intrastat-extrastat-18.0/intrastat-extrastat-18.0-product_harmonized_system
Translate-URL: https://translation.odoo-community.org/projects/intrastat-extrastat-18-0/intrastat-extrastat-18-0-product_harmonized_system/it/
